docs: add a section about the input functions (#408)
hydra-genetics/hydra-genetics: docs/development/input_functions.md

# Input functions

> Hydra Genetics contains helper functions to compile paths to BAM files depending on your pipeline configuration. These paths are used as inputs to some rules in the other Hydra Genetics modules.

Reviewer: ```suggestion
Hydra Genetics contains helper functions to compile paths to BAM files depending on your pipeline configuration. These paths are used  by rules in other Hydra Genetics modules that require aligned BAM files as input.
```
